### PR TITLE
Try to fix cc build

### PR DESCRIPTION
**Pull request by ilyachur**
### Details:
 - Disable warning: unused-local-typedefs

### Tickets:
 - 77710

### CODE REVIEW

File: src/core/shape_inference/CMakeLists.txt [hunk 1]
```diff
@@ -22,6 +22,14 @@ target_include_directories(${TARGET_NAME} PUBLIC
     $<BUILD_INTERFACE:${SHAPE_INFER_INCLUDE_DIR}>
     $<BUILD_INTERFACE:${OV_CORE_INCLUDE_PATH}>)
 
+if (SELECTIVE_BUILD STREQUAL "ON")
+    if(CMAKE_COMPILER_IS_GNUCXX OR OV_COMPILER_IS_CLANG)
```
Comment (ilya-lavrenov) [inline src/core/shape_inference/CMakeLists.txt:26]: we can combine these two `if`

File: src/core/shape_inference/CMakeLists.txt [hunk 2]
```diff
@@ -22,6 +22,14 @@ target_include_directories(${TARGET_NAME} PUBLIC
...
+        # After disabling a block of code, some variables might be unused.
+        target_compile_options(${TARGET_NAME} INTERFACE
+                                -Wno-unused-local-typedefs)
```
Comment (ilya-lavrenov) [inline src/core/shape_inference/CMakeLists.txt:29]: can these typedef be moved under conditionally compiled section?

**Comment by ishariko** [comment]
CI results failed: https://openvino-ci.intel.com/blue/organizations/jenkins/private-ci%2FcustomJobs%2Fbuild-linux/detail/build-linux/242/pipeline
https://openvino-ci.intel.com/job/private-ci/job/customJobs/job/build-linux/242/artifact/b/logs/build/build_ie_for_mobilenet_v2_1.4_224_i8.xml.log

File: src/core/shape_inference/CMakeLists.txt (hunk 2 above)
Comment (ilyachur) [reply src/core/shape_inference/CMakeLists.txt]: Removed

**Review by ilya-lavrenov — APPROVED** [review]

**Pull request merged**